pulsar-flink-connector/src/main/java/org/apache/flink/streaming/connectors/pulsar/FlinkPulsarSinkBase.java

```diff
@@ -279,16 +262,11 @@ public void producerFlush() throws Exception {
                 }
             }
         }
-        synchronized (pendingRecordsLock) {
-            while (pendingRecords > 0) {
-                try {
-                    pendingRecordsLock.wait();
-                } catch (InterruptedException e) {
-                    // this can be interrupted when the Task has been cancelled.
-                    // by throwing an exception, we ensure that this checkpoint doesn't get confirmed
-                    throw new RuntimeException("Flushing got interrupted while checkpointing", e);
-                }
-            }
+
+        long pendingRecordsCount = pendingRecords.get();
```

Comment:
pendingRecords could be removed if we don't require lock.

Reply:
Retaining pendingRecords can ensure that all data has been written out when the snapshot is completed. So there is no need to remove it.